internal/store/models/storedobject.go

```diff
@@ -17,44 +17,38 @@ package models
 
 // StoredObject is the atomic and most abstract description of what is collected by the export store system.
 type StoredObject struct {
-	// id is the unique identifier for this record once stored.
-	id string
+	ID string `bson:"id"`
 
 	// AppServiceKey identifies the app to which this data belongs.
-	AppServiceKey string
+	AppServiceKey string `bson:"appServiceKey"`
 
 	// Payload is the data to be exported
-	Payload []byte
+	Payload []byte `bson:"payload"`
 
 	// RetryCount is how many times this has tried to be exported
-	RetryCount int
+	RetryCount int `bson:"retryCount"`
 
 	// PipelinePosition is where to pickup in the pipeline
-	PipelinePosition int
+	PipelinePosition int `bson:"pipelinePosition"`
 
 	// Version is a hash of the functions to know if the pipeline has changed.
-	Version string
+	Version string `bson:"version"`
 
 	// CorrelationID is an identifier provided by EdgeX to track this record as it moves
-	CorrelationID string
+	CorrelationID string `bson:"correlationID"`
 
 	// EventID is used to identify an EdgeX event from the core services and mark it as pushed.
-	EventID string
+	EventID string `bson:"eventID"`
 
 	// EventChecksum is used to identify CBOR encoded data from the core services and mark it as pushed.
-	EventChecksum string
-}
-
-// GetId returns the unexported field id.
-func (o StoredObject) GetId() string {
-	return o.id
+	EventChecksum string `bson:"eventChecksum"`
 }
 
 // NewStoredObject creates a new instance of StoredObject and is the preferred way to create one.
 func NewStoredObject(id string, appServiceKey string, payload []byte, pipelinePosition int,
```

Comment:
Please remove `id` parameter. When creating a new object, we don't know the ID since it hasn't been saved to the DB.

Reply:
What do you want to use to generate IDs? If we use Mongo's default ObjectID, then a Mongo based store will be incompatible with a Redis based store.

Reply:
I am talking about the `id` parameter. Our code will be calling NewStoredObject. At that time there isn't an `ID` since it hasn't been store to DB.

Reply:
Right. And in my `Store()` function, we cannot use the ID Mongo returns if we want to have interoperability with other DB providers, so we need to define an ID before the store and then use that in the DB.

I'm trying to avoid the problem that EdgeX core services ran into and then solved with Mongo based `models` with just ObjectID IDs, and `contracts` with both a UUID ID and an ObjectID ID.

Reply:
Ahh, I see where we are crossing communication... user should not provide the ID, Store() should generate it similar to how it is generated in edgex-go prior to saving to the DB.